cmd/ackdev/cmd/add_repo.go

```diff
@@ -36,33 +42,38 @@ var addRepositoryCmd = &cobra.Command{
 }
 
 func addRepository(cmd *cobra.Command, args []string) error {
-	for _, service := range args {
-		service = strings.ToLower(args[0])
+	cfg, err := config.Load(ackConfigPath)
+	if err != nil {
+		return err
+	}
 
-		ctx := cmd.Context()
+	repoManager, err := repository.NewManager(cfg)
+	if err != nil {
+		return err
+	}
 
-		cfg, err := config.Load(ackConfigPath)
-		if err != nil {
-			return err
-		}
+	for _, service := range args {
+		service = strings.ToLower(args[0])
 
 		// Check it doesn't already exist in the configuration
 		if util.InStrings(service, cfg.Repositories.Services) {
 			return ErrRepositoryAlreadyAdded
```

Comment:
Rather than failing here, I think I'd actually prefer it to continue to the next service
```suggestion
			fmt.Printf("repository for service %s has already been added\n", service)
			continue
```

Reply:
Agreed ++